mllib/src/main/scala/org/apache/spark/mllib/feature/HashingTF.scala

```diff
@@ -53,9 +65,10 @@ class HashingTF(val numFeatures: Int) extends Serializable {
   @Since("1.1.0")
   def transform(document: Iterable[_]): Vector = {
     val termFrequencies = mutable.HashMap.empty[Int, Double]
+    val setTF = if (binary) (i: Int) => 1.0 else (i: Int) => termFrequencies.getOrElse(i, 0.0) + 1.0
```

Comment:
Yup, the if-else is not actually called within the loop for each term. I think this is ok to keep as is, I don't see any better approach.